packages/react/src/components/List/ListItem/ListItem.jsx

```diff
@@ -234,13 +241,23 @@ const ListItem = ({
   const renderSecondaryValue = () =>
     secondaryValue ? (
       <div
-        title={typeof secondaryValue === 'function' ? `${value}--secondary-value` : secondaryValue}
+        title={
+          typeof secondaryValue === 'function'
+            ? `${value}--secondary-value`
```

Comment:
I think we also need to do something about this, because it never makes sense to show `${value}--secondary-value`. We could check if the function returns a string and if so use that string, otherwise use an empty string.